readers/cassandra/messages.go

```diff
@@ -71,32 +60,15 @@ func (cr cassandraRepository) ReadAll(chanID string, offset, limit uint64, query
 		Messages: []readers.Message{},
 	}
 
-	switch table {
-	case defTable:
-		for scanner.Next() {
-			var msg senml.Message
-			err := scanner.Scan(&msg.Channel, &msg.Subtopic, &msg.Publisher, &msg.Protocol,
-				&msg.Name, &msg.Unit, &msg.Value, &msg.StringValue, &msg.BoolValue,
-				&msg.DataValue, &msg.Sum, &msg.Time, &msg.UpdateTime)
-			if err != nil {
-				return readers.MessagesPage{}, errors.Wrap(errReadMessages, err)
-			}
-			page.Messages = append(page.Messages, msg)
-		}
-	default:
-		for scanner.Next() {
-			var msg jsonMessage
-			err := scanner.Scan(&msg.Channel, &msg.Subtopic, &msg.Publisher, &msg.Protocol, &msg.Created, &msg.Payload)
-			if err != nil {
-				return readers.MessagesPage{}, errors.Wrap(errReadMessages, err)
-			}
-			m, err := msg.toMap()
-			if err != nil {
-				return readers.MessagesPage{}, errors.Wrap(errReadMessages, err)
-			}
-			m["payload"] = jsont.ParseFlat(m["payload"])
-			page.Messages = append(page.Messages, m)
+	for scanner.Next() {
+		var msg senml.Message
+		err := scanner.Scan(&msg.Channel, &msg.Subtopic, &msg.Publisher, &msg.Protocol,
```

Comment:
This is probably more readable, but I am not particularly against embedding this in the error check.